### DISCUSSION TITLE
Is there a way to add method to parsed data?

### DISCUSSION THREAD

**Discussion by hellozyemlya** [colinhacks/zod#3074] [Q&A] [answered]
Lets assume i have a {a: 1, b: 2} data, and after parsing I want to add method sum to object. Is that possible with zod?

**Comment by JacobWeisenburger** [comment]
Is this what you are looking for?
https://github.com/colinhacks/zod#transform

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by hellozyemlya** [reply]
Not really, I expect to have object like {a: 1, b:1, sum: ()=>{}}, z.custom with .default looks like I wanted with single exception - I don't have access to a and b values in my function.

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/colinhacks/zod#transform

Is this what you are looking for?

```ts
const schema = z.object( {
    a: z.number(),
    b: z.number(),
} ).transform( data => ( {
    ...data,
    sum: () => data.a + data.b
} ) )

type Input = z.input<typeof schema>
// type Input = {
//     a: number
//     b: number
// }

type Output = z.output<typeof schema>
// type Output = {
//     a: number
//     b: number
//     sum: () => number
// }

console.log( schema.parse( { a: 1, b: 2 } ) )
// {
//     a: 1,
//     b: 2,
//     sum: [Function: sum]
// }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by hellozyemlya** [reply]
Hm, will try it out. But looks like very close... And seems like I can hack object as I want and create fully mutable object. Anyway, thanks a lot for help.
